packages/flutter_tools/lib/src/commands/analyze.dart

```diff
@@ -127,13 +129,25 @@ class AnalyzeCommand extends FlutterCommand {
 
   @override
   Future<FlutterCommandResult> runCommand() async {
-    final String? suggestionFlag = stringArg('suggestions');
-    if (suggestionFlag != null) {
+    final bool? suggestionFlag = boolArg('suggestions');
+    final Set<String> items;
+    final String directoryPath;
+    if (suggestionFlag != null && suggestionFlag == true) {
```

Comment:
can you add a check for if the user provided BOTH `--suggestions` AND `--watch`? I'm guessing we should just tool exit and tell them they're not compatible. If we don't fail when both are provided, then one overrides the other in an undocumented way.